[personalization_ext] Improve personalization extensions API

vbendeb opened the pull request:
Declare interface functions in a .h file so that they can be validated at compile time and move all passed in parameters into the respective structures. Not all extesnions use all interface parameters, keeping paramters in structures will allow to add new interface parameters without modifying extensions which do not use the new parameters.

Tested by running various personalization end to end tests both local and with extensions.

vbendeb commented:
A noop change improving personalization extension API robustness.

vbendeb commented:
@tommychiu-github somehow I can't add Tommy to the reveiwers list explicitly, maybe this works.

timothytrippel commented on sw/device/silicon_creator/manuf/extensions/example_personalize_ext.c:
```diff
@@ -8,6 +8,7 @@
 #include "sw/device/lib/testing/test_framework/status.h"
 #include "sw/device/lib/testing/test_framework/ujson_ottf.h"
 #include "sw/device/silicon_creator/lib/cert/cert.h"
+#include "sw/device/silicon_creator/manuf/base/personalize_ext.h"
```
I think you need to modify the inputs to the two perso extension functions in this file too.

vbendeb replied:
done.

timothytrippel commented on sw/device/silicon_creator/manuf/base/BUILD:
```diff
@@ -214,6 +214,7 @@ filegroup(
 cc_library(
     name = "ft_personalize_base",
     srcs = ["ft_personalize.c"],
+    hdrs = ["personalize_ext.h"],
```
I think what you really want to do is make the header file its own lib like:
```
cc_library(
    name = "personalize_ext",
    hdrs = ["personalize_ext.h"],
)
```

Then include it as a dep of `ft_personalize_base` and all the extension libs.

vbendeb replied:
done

timothytrippel commented on sw/device/silicon_creator/manuf/base/BUILD:
```diff
@@ -251,7 +252,9 @@ cc_library(
 cc_library(
     name = "tpm_perso_fw_ext",
     srcs = ["tpm_personalize_ext.c"],
+    hdrs = ["personalize_ext.h"],
     deps = [
+        ":ft_personalize_base",
```
If you make my suggested change above, then this would be `personalize_ext`, and you would not need the hdrs bit above, like so:
```suggestion
    deps = [
        ":personalize_ext",
```

vbendeb replied:
done

timothytrippel commented on sw/device/silicon_creator/manuf/extensions/BUILD.bazel:
```diff
@@ -29,6 +29,8 @@ cc_library(
         "@//sw/device/lib/testing/test_framework:status",
         "@//sw/device/lib/testing/test_framework:ujson_ottf",
         "@//sw/device/silicon_creator/lib/cert",
+        "@//sw/device/silicon_creator/manuf/base:ft_personalize_base",
+        "@//sw/device/silicon_creator/manuf/lib:flash_info_fields",
```
After making the new cc_lib I mention above, then do:
```suggestion
        "@//sw/device/silicon_creator/manuf/base:personalize_ext",
```

vbendeb replied:
I thought I included flash_info_fields for a reason, but seems to be not necessary...

done

timothytrippel commented on sw/device/silicon_creator/manuf/extensions/BUILD.bazel:
```diff
@@ -41,6 +43,8 @@ cc_library(
         "@//sw/device/lib/testing/test_framework:status",
         "@//sw/device/lib/testing/test_framework:ujson_ottf",
         "@//sw/device/silicon_creator/lib/cert",
+        "@//sw/device/silicon_creator/manuf/base:ft_personalize_base",
+        "@//sw/device/silicon_creator/manuf/lib:flash_info_fields",
```
same as above

vbendeb replied:
done

timothytrippel commented on sw/device/silicon_creator/manuf/base/personalize_ext.h:
```diff
@@ -0,0 +1,87 @@
+  /**
+   * Pointer to the flash layout table, the extension could enable disabled
+   * entries in the table and configure its INFO space use so that the caller
+   * knows where to endorsed objects received from the host.
```
```suggestion
   * knows where to place endorsed objects received from the host.
```

vbendeb replied:
done

timothytrippel commented on sw/device/silicon_creator/manuf/base/personalize_ext.h:
```diff
@@ -0,0 +1,87 @@
+  perso_blob_t *perso_blob_from_host;
+  /**
+   * Pointer to the flash layout table used previously when saving pre
+   * indorsement data..
```
```suggestion
   * endorsement data.
```

tommychiu-github commented:
> @tommychiu-github somehow I can't add Tommy to the reveiwers list explicitly, maybe this works.

Thanks for sharing me this change.
I'll need to update pixel sku to adopt this mechanism as well.

vbendeb commented:
PTAL (and I understand that if I change an object in a BUILD file I might have to edit other build files ;)

timothytrippel reviewed (approved):
LGMT, thanks @vbendeb

timothytrippel commented:
Looks like you have some lint errors though.

vbendeb commented:
> Looks like you have some lint errors though.

I could find only one error even though the builder shows two. One of the other logs seems completely innocuous. 

I now added a pre-upload check for include guards, updated, let's see if there are more lint issues.
